src/order/basic.lean

```diff
@@ -120,6 +120,12 @@ lemma reflect_lt {α β} [linear_order α] [preorder β] {f : α → β} (hf : m
   {x x' : α} (h : f x < f x') : x < x' :=
 by { rw [← not_le], intro h', apply not_le_of_lt h, exact hf h' }
 
+/-- If `f` is a monotone function from `ℕ` to a preorder such that `y` lies between `f x` and
+  `f (x + 1)`, then `y` doesn't lie in the range of `f`. -/
+lemma ne_of_monotone_of_lt_of_lt {α} [preorder α] {f : ℕ → α} (hf : monotone f)
```

Comment:
This is already inside the `monotone` namespace.
```suggestion
lemma ne_of_lt_of_lt {α} [preorder α] {f : ℕ → α} (hf : monotone f)
```

Reply:
While you're at it, could you please move `reflect_lt` to the `monotone` namespace?

Reply:
Yes, that is better. I also added a version for `int`.